File: src/dataset/javascript/say_exercism-ThomasZumsteg/gemini/GeminiFlash_say_v2.js
const ones = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
const teens = ["", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"];
const tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"];
const powers = ["", " thousand", " million", " billion"];

function inEnglish(num) {
  if (num < 0 || num >= 1000000000000) {
    throw new Error('Number must be between 0 and 999,999,999,999.');
  }

  if (num === 0) {
    return "zero";
  }

  let english = "";
  let power = 0;

  while (num > 0) {
    const group = num % 1000;
    if (group > 0) {
      english = englishPower(group) + powers[power] + (english ? " " : "") + english;
    }
    num = Math.floor(num / 1000);
    power++;
  }

  return english;
}

function englishPower(digits) {
  let english = "";
  const hundred = Math.floor(digits / 100);
  const remainder = digits % 100;

  if (hundred > 0) {
    english += ones[hundred] + " hundred";
  }

  if (remainder > 0) {
    if (english !== "") {
      english += " ";
    }

    if (remainder < 10) {
      english += ones[remainder];
    } else if (remainder < 20) {
      english += teens[remainder - 10];
    } else {
      const ten = Math.floor(remainder / 10);
      const one = remainder % 10;
      english += tens[ten];
      if (one > 0) {
        english += "-" + ones[one];
      }
    }
  }

  return english;
}

module.exports = { inEnglish: inEnglish };

File: src/dataset/python/say_exercism-ThomasZumsteg/gemini/GeminiFlash_say_v2.py
NUMBERS = [''] + 'one two three four five six seven eight nine'.split()
TENS = [''] + 'ten twenty thirty forty fifty sixty seventy eighty ninety'.split()
TEENS = 'ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen'.split()

POWERS = ['', ' thousand', ' million', ' billion']

def say(num):
    """say constructs the english phrase for a number between 0 and one trillion"""

    if not (0 <= num < 1E12):
        raise AttributeError("Number must be between 0 and one trillion")
    if num == 0:
        return 'zero'

    parts = []
    for i in range(4):
        num, power = divmod(num, 1000)
        if power:
            parts.append(say_power(power) + POWERS[i])

    return ' '.join(parts[::-1]).strip()


def say_power(num):
    """say_power converts a number between 1 and 999 to the english phrase"""
    hundreds = num // 100
    tens = (num % 100) // 10
    ones = num % 10

    parts = []
    if hundreds:
        parts.append(NUMBERS[hundreds] + ' hundred')
        if tens or ones:
            parts.append('and')

    if tens == 1:
        parts.append(TEENS[ones])
    else:
        if tens:
            parts.append(TENS[tens])
        if ones:
            if tens:
                parts.append('-' + NUMBERS[ones])
            else:
                parts.append(NUMBERS[ones])

    return ' '.join(parts)